matchms/similarity/ModifiedCosine.py

```diff
@@ -0,0 +1,102 @@
+import numba
+import numpy
+from matchms.typing import SpectrumType
+
+
+class ModifiedCosine:
+    """Calculate modified cosine score between spectra.
+
+    This score is calculated based on matches between peaks of two spectra and
+    also includes a shift of peak positions by the difference in precursor mass.
+
+    Args:
+    ----
+    tolerance: float
+        Peaks will be considered a match when <= tolerance appart. Default is 0.1.
+    """
+    def __init__(self, tolerance=0.1):
+        self.tolerance = tolerance
+
+    def __call__(self, spectrum1: SpectrumType, spectrum2: SpectrumType) -> float:
```

Comment:
It returns score and nr matches
```suggestion
    def __call__(self, spectrum1: SpectrumType, spectrum2: SpectrumType) -> Tuple[float, int]:
```